packages/types/src/checksum.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * An object that provides a checksum of data provided in chunks to `update`.
+ * The checksum may be performed incrementally as chunks are received or all
+ * at once when the checksum is finalized, depending on the underlying
+ * implementation.
+ *
+ * It's recommended to compute checksum incrementally to avoid reading the
+ * entire payload in memory.
+ *
+ * A class that implements this interface may accept an optional secret key in its
+ * constructor while computing checksum value, when using HMAC. If provided,
+ * this secret key would be used when computing checksum.
+ */
+export interface Checksum {
+  /**
+   * Constant length of the digest created by the algorithm in bytes.
+   */
+  digestLength?: number;
+
+  /**
+   * Creates a new checksum object that contains a deep copy of the internal
+   * state of the current `Checksum` object.
+   */
+  copy?: () => Checksum;
+
+  /**
+   * Returns the digest of all of the data passed.
+   */
+  digest: () => Promise<Uint8Array>;
+
+  /**
+   * Allows marking a checksum for checksums that support the ability
+   * to mark and reset.
+   *
+   * @param {number} readLimit - The maximum limit of bytes that can be read
+   *   before the mark position becomes invalid.
+   */
+  mark?: (readLimit: number) => void;
+
+  /**
+   * Resets the checksum to its initial value.
+   */
+  reset: () => void;
```

Comment:
Use method syntax in interface
```suggestion
  reset(): void;
```